src/KubernetesClient/StreamDemuxer.cs

```diff
@@ -199,6 +199,13 @@ protected async Task RunLoop(CancellationToken cancellationToken)
 
                     var streamIndex = buffer[0];
                     var extraByteCount = 1;
+                    byteCount += result.Count - extraByteCount;
```

Comment:
I didn't realize StreamDemuxer is used in other scenarios as well. Added enum StreamType, default to RemoteCommand (pick the name RemoteCommand since exec and attach are implemented under kubelet\server\remotecommand directory). StreamType.RemoteCommand does not perform extra handling. If StreamType.PortForward is specified, StreamDemuxer will filter out the port bytes.